clippy_lints/src/doc.rs

```diff
@@ -516,34 +518,54 @@ fn check_doc<'a, Events: Iterator<Item = (pulldown_cmark::Event<'a>, Range<usize
             Html(_html) => (),             // HTML is weird, just ignore it
             SoftBreak | HardBreak | TaskListMarker(_) | Code(_) | Rule => (),
             FootnoteReference(text) | Text(text) => {
-                if Some(&text) == in_link.as_ref() {
+                if Some(&text) == in_link.as_ref() || ticks_unbalanced {
                     // Probably a link of the form `<http://example.com>`
                     // Which are represented as a link to "http://example.com" with
                     // text "http://example.com" by pulldown-cmark
                     continue;
                 }
-                headers.safety |= in_heading && text.trim() == "Safety";
-                headers.errors |= in_heading && text.trim() == "Errors";
-                headers.panics |= in_heading && text.trim() == "Panics";
-                let index = match spans.binary_search_by(|c| c.0.cmp(&range.start)) {
-                    Ok(o) => o,
-                    Err(e) => e - 1,
-                };
-                let (begin, span) = spans[index];
-                if in_code {
-                    if is_rust {
-                        let edition = edition.unwrap_or_else(|| cx.tcx.sess.edition());
-                        check_code(cx, &text, edition, span);
-                    }
+                if text.contains('`') && !in_code {
+                    let (_, start) = spans[0];
```

Comment:
Just a NIT: You can use [`clippy_util::line_span(..)`](https://github.com/rust-lang/rust-clippy/blob/d18441383362c1a55f72949677f972a2c42972c2/clippy_utils/src/lib.rs#L786-L801) to expand the message to the entire line. That will make the message a bit cleaner in this case.

```txt
// It should change this:
LL |   /// This is a doc comment with `unbalanced_tick marks and several words that
   |  ____^
LL | | /// should be `encompassed_by` tick marks because they `contain_underscores`.

// To only show the lines without a start indication:
LL |   /// This is a doc comment with `unbalanced_tick marks and several words that
LL |   /// should be `encompassed_by` tick marks because they `contain_underscores`.
```